Riot/Utils/EventFormatter.m

```diff
@@ -169,52 +169,119 @@ - (NSAttributedString *)attributedStringFromEvent:(MXEvent *)event withRoomState
         }
     }
     
-    if (event.eventType == MXEventTypeRoomCreate)
+    switch (event.eventType)
     {
-        MXRoomCreateContent *createContent = [MXRoomCreateContent modelFromJSON:event.content];
-        
-        NSString *roomPredecessorId = createContent.roomPredecessorInfo.roomId;
-        
-        if (roomPredecessorId)
+        case MXEventTypeRoomCreate:
         {
-            return [self roomCreatePredecessorAttributedStringWithPredecessorRoomId:roomPredecessorId];
+            MXRoomCreateContent *createContent = [MXRoomCreateContent modelFromJSON:event.content];
+            
+            NSString *roomPredecessorId = createContent.roomPredecessorInfo.roomId;
+            
+            if (roomPredecessorId)
+            {
+                return [self roomCreatePredecessorAttributedStringWithPredecessorRoomId:roomPredecessorId];
+            }
+            else
+            {
+                NSAttributedString *string = [super attributedStringFromEvent:event withRoomState:roomState error:error];
+                NSMutableAttributedString *result = [[NSMutableAttributedString alloc] initWithString:@"· "];
+                [result appendAttributedString:string];
+                return result;
+            }
         }
-        else
+            break;
+        case MXEventTypeRoomRetention:
         {
-            NSAttributedString *string = [super attributedStringFromEvent:event withRoomState:roomState error:error];
-            NSMutableAttributedString *result = [[NSMutableAttributedString alloc] initWithString:@"· "];
-            [result appendAttributedString:string];
-            return result;
+            // Check whether a retention period is defined
+            uint periodInDays = RoomService.undefinedRetentionValueInDays;
+            if (event.content[RoomService.roomRetentionContentMaxLifetimeKey])
+            {
+                UInt64 maxLifetime = -1;
+                MXJSONModelSetUInt64(maxLifetime, event.content[RoomService.roomRetentionContentMaxLifetimeKey]);
+                periodInDays = [Tools numberOfDaysFromDurationInMs:maxLifetime];
```

Comment:
-1 signed = 18446744073709551615 unsigned = 0xFFFFFFFFFFFFFFFF hexa

```suggestion
                MXJSONModelSetUInt64(UINT64_MAX, event.content[RoomService.roomRetentionContentMaxLifetimeKey]);
                periodInDays = [Tools numberOfDaysFromDurationInMs:UINT64_MAX];
```

Reply:
You're right I should use UINT64_MAX instead of -1
but I should keep use : UInt64 maxLifetime = UINT64_MAX;